include/flux/op/find.hpp

```diff
@@ -6,44 +6,71 @@
 #ifndef FLUX_OP_FIND_HPP_INCLUDED
 #define FLUX_OP_FIND_HPP_INCLUDED
 
+#include <flux/core/utils.hpp>
 #include <flux/op/for_each_while.hpp>
 
+#include <cstring>
+#include <type_traits>
+
 namespace flux {
 
 namespace detail {
 
 struct find_fn {
+private:
+    template <sequence Seq, typename Value>
+    constexpr auto impl(Seq&& seq, Value const& value) const -> cursor_t<Seq>
+    {
+        return for_each_while(
+            seq, [&](auto&& elem) { return FLUX_FWD(elem) != value; });
+    }
+
+public:
     template <sequence Seq, typename Value>
         requires std::equality_comparable_with<element_t<Seq>, Value const&>
-    constexpr auto operator()(Seq&& seq, Value const& value) const -> cursor_t<Seq>
+    constexpr auto operator()(Seq&& seq, Value const& value) const
+        -> cursor_t<Seq>
     {
-        return for_each_while(seq, [&](auto&& elem) {
-            return FLUX_FWD(elem) != value;
-        });
+        constexpr auto memchr_opt = flux::contiguous_sequence<Seq> &&
+            flux::sized_sequence<Seq> && std::is_integral_v<Value> &&
+            flux::detail::any_of<value_t<Seq>, char, signed char, unsigned char,
+                const char, const signed char, const unsigned char>;
+
+        if constexpr (memchr_opt) {
+            if (std::is_constant_evaluated()) {
+                return impl(seq, value);
+            }
+            auto location = std::memchr(flux::data(seq), value,
+                flux::usize(seq) * sizeof(value_t<Seq>));
+            if (location == nullptr)
+                return flux::usize(seq);
+            else
+                return static_cast<const value_t<Seq>*>(location) -
+                    flux::data(seq);
```

Comment:
Braces on if statements, please :)

This isn't quite returning the right thing. `flux::find` returns a *cursor*, which for a contiguous sequences is an integer index -- but for subsequences, the first element might not have index zero, so we have to adjust the return value. This is what is causing the test failures with `split`, because it calls `find()` on a subsequence, hits the new specialisation and gets the wrong answer.

```suggestion
            if (location == nullptr) {
                return flux::last(seq);
            } else {
                auto offset = static_cast<value_t<Seq> const*>(location) - flux::data(seq);
                return flux::next(seq, flux::first(seq), offset);
            }
```

Reply:
Thanks for this! I made the changes you suggested but I'm still getting a failure on the unit tests. I'll have to dig in deeper.

Reply:
Disregard, that was my mistake.